src/programs.rkt

```diff
@@ -332,7 +332,10 @@
 
 (define (resugar-program prog prec)
   (define repr (get-representation prec))
-  (expand-parametric-reverse prog repr))
+  (match prog
+    [(list 'FPCore (list vars ...) body) `(FPCore ,vars ,(expand-parametric-reverse body repr))]
+    [(list 'λ (list vars ...) body) `(λ ,vars ,(expand-parametric-reverse body repr))]
```

Comment:
It's convenient (for debugging) to allow both `λ` and `lambda`. I'd write the pattern `(list (or 'FPCore 'λ 'lambda) (list vars ...) body)`.

Reply:
Good to know. Thanks. I'll fix then merge